rudder-core/src/main/scala/com/normation/rudder/services/system/DatabaseManager.scala

```diff
@@ -111,14 +121,14 @@ class DatabaseManagerImpl(
    }
 
 
-   def deleteEntries(date : DateTime) : Box[Int] = {
-     val reports = reportsRepository.deleteEntries(date) ?~! "An error occured while deleting reports"
-     val nodeConfigs = expectedReportsRepo.deleteNodeConfigIdInfo(date) ?~! "An error occured while deleting old node configuration IDs"
-     val deleteNodeConfigs = expectedReportsRepo.deleteNodeConfigurations(date) ?~! "An error occured while deleting Node Configurations"
-     val deleteNodeCompliances = expectedReportsRepo.deleteNodeCompliances(date) ?~! "An error occured while deleting Node Compliances"
+   def deleteEntries(reports: DeleteCommand.Reports, complianceLevels: Option[DeleteCommand.ComplianceLevel]) : Box[Int] = {
+     val nodeReports = reportsRepository.deleteEntries(reports.date) ?~! "An error occured while deleting reports"
+     val nodeConfigs = expectedReportsRepo.deleteNodeConfigIdInfo(reports.date) ?~! "An error occured while deleting old node configuration IDs"
+     val deleteNodeConfigs = expectedReportsRepo.deleteNodeConfigurations(reports.date) ?~! "An error occured while deleting Node Configurations"
+     val deleteNodeCompliances = expectedReportsRepo.deleteNodeCompliances(reports.date) ?~! "An error occured while deleting Node Compliances"
+     val deleteNodeComplianceLevels = Box(complianceLevels).flatMap(c => expectedReportsRepo.deleteNodeComplianceLevels(c.date) ?~! "An error occured while deleting Node Compliances")
```

Comment:
This part fails when we disable deletion of complianceLevels as it converts to Empty